subt/tools/draw_pos_data.py

```diff
@@ -30,20 +31,71 @@ def read_all(folder):
     return ret
 
 
-def draw(robots):
+def draw(robots, create_video_path=None):
+    fig, ax = plt.subplots()
+    size = 0
     for name, arr in robots.items():
         x = [xyz[0] for t, xyz in arr]
         y = [xyz[1] for t, xyz in arr]
         plt.plot(x, y, '-', label=name)
-    plt.axes().set_aspect('equal', 'datalim')
+        size = max(size, len(x))
+    ax.set_aspect(1.0)
     plt.legend()
+
+    # The function to be called anytime a slider's value changes
+    def update(val):
+        ax.clear()
+        for name, arr in robots.items():
+            x = [xyz[0] for t, xyz in arr]
+            y = [xyz[1] for t, xyz in arr]
+            index = min(int(val), len(x) - 1)
+            ax.plot(x[:index+1], y[:index+1], '-', label=name)
+            ax.scatter([x[index]], [y[index]], s=50)
+        fig.canvas.draw_idle()
+
+    if create_video_path is not None:
+        assert create_video_path.endswith(".mp4"), create_video_path
+
+        tmp_image = "tmp_img.png"
+        writer = None
+        fps = 10
+        for i in range(size):
+            update(i)
+            plt.savefig(tmp_image)
```

Comment:
If you use [`tempfile.TemporaryFile`](https://docs.python.org/3/library/tempfile.html), it will get nicely delete after you are done.

Alternatively, and maybe even better, it seems you could use `io.BytesIO` instead: https://stackoverflow.com/questions/8598673/how-to-save-a-pylab-figure-into-in-memory-file-which-can-be-read-into-pil-image

Reply:
OK, thanks, used also with help of
https://stackoverflow.com/questions/46624449/load-bytesio-image-with-opencv